var/spack/repos/builtin/packages/amdscalapack/package.py

```diff
@@ -20,19 +20,34 @@ class Amdscalapack(ScalapackBase):
 
     _name = 'amdscalapack'
     homepage = "https://developer.amd.com/amd-aocl/scalapack/"
-    url = "https://github.com/amd/scalapack/archive/3.0.tar.gz"
     git = "https://github.com/amd/scalapack.git"
 
     maintainers = ['amd-toolchain-support']
 
+    version('3.1', sha256='4c2ee2c44644a0feec0c6fc1b1a413fa9028f14d7035d43a398f5afcfdbacb98')
     version('3.0', sha256='6e6f3578f44a8e64518d276e7580530599ecfa8729f568303ed2590688e7096f')
     version('2.2', sha256='2d64926864fc6d12157b86e3f88eb1a5205e7fc157bf67e7577d0f18b9a7484c')
 
+    def url_for_version(self, version):
+        if version == Version('3.1'):
+            return "https://github.com/amd/aocl-scalapack/archive/3.1.tar.gz"
+        elif version == Version('3.0'):
+            return "https://github.com/amd/scalapack/archive/3.0.tar.gz"
+        elif version == Version('2.2'):
+            return "https://github.com/amd/scalapack/archive/2.2.tar.gz"
+
```

Comment:
It is preferable to have the directives listed *before* methods so can this be moved to appear after the `conflicts` directive so it is included in the methods "section"?

Reply:
Incorporated review comments